pwiz_tools/Skyline/Model/AuditLog/AuditLogEntry.cs

```diff
@@ -1224,17 +1224,18 @@ public byte[] GetAuditLogHash()
             using (var sha1 = new SHA1CryptoServiceProvider())
             {
                 var enc = Encoding.UTF8;
-                BlockHash blockHash = new BlockHash(sha1);
+                var blockHash = new BlockHash(sha1);
                 blockHash.ProcessBytes(enc.GetBytes(User));
                 if (!string.IsNullOrEmpty(ExtraInfo))
+                {
                     blockHash.ProcessBytes(enc.GetBytes(LogMessage.ParseLogString(ExtraInfo, LogLevel.all_info,
                         CultureInfo.InvariantCulture)));
+                }
 
                 blockHash.ProcessBytes(enc.GetBytes(UndoRedo.ToString(CultureInfo.InvariantCulture)));
                 blockHash.ProcessBytes(enc.GetBytes(Summary.ToString(CultureInfo.InvariantCulture)));
                 _allInfoNoUndoRedo.ForEach(l => blockHash.ProcessBytes(enc.GetBytes(l.ToString(CultureInfo.InvariantCulture))));
-                // TODO: include other information in the hash?
```

Comment:
I believe there is reason element in the audit_log_entry schema. I think it should be included into the hash as well.

Reply:
Good catch, added that to the hash. The reason will be hashed right after its corresponding message. FYI, there is only one reason for the undo redo message, the summary message, and the first all info message (which is simply the undo redo message) and all remaining all info message can have their own reasons.